### DISCUSSION TITLE
Limit or warn over continous usage/cooldown

### DISCUSSION THREAD

**Discussion by RodrigoDiazDVivar** [proginosko/LeechBlockNG#366] [Ideas]
I have the issue that when I allow a page for, eg "30 minutes a day" and I start using it at 23:40; i end up being on the site for 50 minutes straight and dont even realise because the counter has reseted!!

I suggest adding an "allow cooldown of (insert) minutes" option that blocks the site for a specified cooldown time if you use it more than the specified allowed time (in my example 30 minutes) between two consecutive periods. In my example, for instance, it should be blocked at 00:10 given that I started at 23:40. If I chose a cooldown of 5 minutes, that would mean that the site would remain blocked until 00:15. 

Or perhaps, more easily you could just add an intrusive alert that says:
<< WARNING: THE COUNTDOWN HAS BEEN RESETED DUE TO CHANGE IN PERIOD. YOU HAVE USED THIS SITE FOR *** MINUTES IN THE LAST PERIOD (OR CONTINOUSLY UP TO NOW). CLICK "OK" TO CLOSE THIS MESSAGE>>
 or something alike, although it wouldn't respect my original idea of allowing up to "..... minutes" in a row.

Another option would be adding a maximum time "in a row" and then force a "cooldown time" afterword: Eg allow up to 2 minutes in a row with a cooldown of 30 minutes after you hit the 2 min mark.

Thank you!

**Comment by Armnoo** [comment]
Hello! Did you found a workaround for the idea. I want a cooldown of 2 hours after 30 min of use.

**Reply by RodrigoDiazDVivar** [reply]
Not really, tbh i dont use this addon a lot anymore. It's better to just be aware. In your case you could maybe make "blocks" of 30 minutes allowed every 2:30 hours. It's not the same but i guess it's the best you can do

**Reply by Armnoo** [reply]
Oh well thanks for the reply

**Reply by proginosko** [reply]
I plan to add an option to specify a minimum duration for the block (e.g., "Always block sites for at least __ minutes"). That should provide the functionality you want.

**Comment by RodrigoDiazDVivar** [comment]
So that's at the beginning?

**Comment by proginosko** [comment]
Added option "Always block sites for at least __ minutes" in version 1.7.2.
